src/vst/ivsttestplugprovider.rs

```diff
@@ -0,0 +1,13 @@
+use crate::base::tresult;
+use crate::vst::{IComponent, IEditController, IStringResult};
+use vst3_com::com_interface;
+use vst3_com::interfaces::iunknown::IUnknown;
+
+#[com_interface("BAB58FD6-8F97-6E1E-4E99-430F86BE70EE")]
+pub trait ITestPlugProvider: IUnknown {
+    unsafe fn get_component(&self) -> Option<*mut IComponent>;
+    unsafe fn get_controller(&self) -> Option<*mut IEditController>;
```

Comment:
return *mut dyn IEditController instead of an option

Reply:
@m-hilgendorf Fat pointers don't behave well across language boundaries, these should return *mut c_void pointers as I wrote some time ago here #23

Reply:
Good point! For the moment returning fat pointers "works" in that they can be cast to `ComPtr` or `ComRc` for dispatching to methods, like we do with the `IParameterChanges` wrapping in the examples.

I think to be more explicit we should return these as pointers to vtables, which is how `ComPtr` works under the hood, e.g. 

```
unsafe fn get_compent(&self) -> *mut <dyn IComponent as ComInterface>::VTable
```

Reply:
That should work and it would be more explicit because it shows the specific Trait name returned! Very good!